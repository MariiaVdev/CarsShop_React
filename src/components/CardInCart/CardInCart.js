import styles from './CardInCart.module.scss';
import PropTypes from 'prop-types';


const CardInCart = (props) => {
        const { name, price, img, code, color} = props;
        return (

            <div className={styles.cardWrapper}>
                <img src={img} alt={name}></img>
                <div className={styles.cardName}><h3>{name}
                </h3></div>
                <div className={styles.cardCode}>Vendor code: {code}</div>
                <div className={styles.cardCode}>Color: <button className={styles.cardBtnColor} style={{ backgroundColor: color }}></button></div>
                <div className={styles.cardPrice}><p><b>${price}</b></p>
                </div>
            </div>

        )
}

CardInCart.propTypes = {
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    img: PropTypes.string.isRequired,
    code: PropTypes.number,
    color: PropTypes.string

}

CardInCart.defaultProps = {
    code: 1234,
    color: 'grey'
}

export default CardInCart;