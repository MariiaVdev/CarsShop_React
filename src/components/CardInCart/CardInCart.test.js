import { render } from "@testing-library/react"
import CardInCart from './CardInCart'


describe("Render CardInCart", () => {
    test("should CardInCart render with default props", () => {
        const { asFragment } = render(<CardInCart name={'test'} price={12345} img={'img'} />);
        expect(asFragment()).toMatchSnapshot()
    })
    test("should CardInCart render with custom props", () => {
        const { asFragment } = render(<CardInCart name={'test'} price={12345} img={'img'} code={1234} color={'red'} />);
        expect(asFragment()).toMatchSnapshot()
    })
})


