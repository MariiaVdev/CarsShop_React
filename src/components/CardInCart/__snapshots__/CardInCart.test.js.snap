// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CardInCart should CardInCart render with custom props 1`] = `
<DocumentFragment>
  <div
    class="cardWrapper"
  >
    <img
      alt="test"
      src="img"
    />
    <div
      class="cardName"
    >
      <h3>
        test
      </h3>
    </div>
    <div
      class="cardCode"
    >
      Vendor code: 1234
    </div>
    <div
      class="cardCode"
    >
      Color: 
      <button
        class="cardBtnColor"
        style="background-color: red;"
      />
    </div>
    <div
      class="cardPrice"
    >
      <p>
        <b>
          $12345
        </b>
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Render CardInCart should CardInCart render with default props 1`] = `
<DocumentFragment>
  <div
    class="cardWrapper"
  >
    <img
      alt="test"
      src="img"
    />
    <div
      class="cardName"
    >
      <h3>
        test
      </h3>
    </div>
    <div
      class="cardCode"
    >
      Vendor code: 1234
    </div>
    <div
      class="cardCode"
    >
      Color: 
      <button
        class="cardBtnColor"
        style="background-color: grey;"
      />
    </div>
    <div
      class="cardPrice"
    >
      <p>
        <b>
          $12345
        </b>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
